src/run.rs

```diff
@@ -66,22 +61,26 @@ pub trait Handler {
     /// - `Err`: an error has occurred while processing, quit.
     /// - `Ok(true)`: everything is fine and the loop can continue.
     /// - `Ok(false)`: everything is fine but we should gracefully stop.
-    fn on_update(&mut self, ops: &[PathOp]) -> Result<bool>;
+    fn on_update(&self, ops: &[PathOp]) -> Result<bool>;
+
+    /// Handler implementations must return a customized watchexec Args reference for use in the
+    /// calling watcher.
+    fn args(&self) -> &Args;
```

Comment:
If you could amend the documentation in the style of the other methods... describing when (at start of watch) and how often (once) it's called especially